### PR TITLE
wip: chore(cherow): Optimization for tokenizing string literal (fixes #117)

### LINKED ISSUE #117: Optimization for tokenizing string literal.
We need to optimize the tokenizer, with string literals and templates.
Do some refactors, replace `String.fromCharCode` with string slices.

### PR DESCRIPTION
**Pull request by Aladdin-ADD**
let me know if some more can be done. 😄

### CODE REVIEW

File: packages/cherow/src/lexer/string.ts [hunk 1]
```diff
@@ -16,41 +16,35 @@ import { Errors, report } from '../errors';
...
+              ret += parser.source[parser.index];
       }
-
-      ch = readNext(parser, ch);
   }
 
   parser.index++;
   parser.column++; // Consume the quote
   parser.tokenRaw = parser.source.slice(start, parser.index);
   parser.tokenValue = ret;
```
Comment (Aladdin-ADD) [inline packages/cherow/src/lexer/string.ts]: sorry, I'm not sure I got it correctly. Do you mean adding something like `parser.lastValue = ret.slice(...);` here? but I'm not seeing `parser.lastValue` used, and all tests passed.

Comment (KFlash) [reply packages/cherow/src/lexer/string.ts]: Look into the table lookup and you will see it's used. Plus it's used when handling the returned error messages. Btw. I tried to slice before, but didn't work out 100%. If you look I guess 8 commits on the 2.0 branch you will see how I tried to do it to better understand.

**Comment by KFlash** [comment]
This pull request **introduces 3 alerts** when merging b5594e219b836e7f462b5f56a6802d2ab63109a5 into 0b21c5022782d1e0160ec68927a7b99ba92ce561 - [view on LGTM.com](https://lgtm.com/projects/g/cherow/cherow/rev/pr-6da0fc4a99761ed9d1d046da8bc6584163dc68dd)

**new alerts:**

* 3 for Useless assignment to local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by KFlash** [comment]
@Aladdin-ADD your latest changes looks promising, but closed because moving to Master brach. Can you open a new PR targeting Master?

**Pull request closed without merging**